<launch>
  <arg name="launch_websocket" default="true" />
  <arg name="debug" default="false" />

  <param name="mongodb_use_daemon" value="true" />
  <param name="mongodb_host" value="localhost" />
  <param name="mongodb_port" value="27017" />
  <node name="message_store" pkg="mongodb_store" type="message_store_node.py" output="screen" required="true" /> 

  <include if="$(arg launch_websocket)" file="$(find rosbridge_server)/launch/rosbridge_websocket.launch" />

  <rosparam command="load" file="$(find task_perception)/config/particle_tracker.yaml" />
  <rosparam command="load" file="$(find task_perception)/config/camera.yaml" />
  <rosparam command="load" file="$(find task_perception)/config/object.yaml" />
  <node pkg="dbot_ros" type="multi_object_tracker_node" name="multi_object_tracker_server" />
  <node pkg="dbot_ros" type="initialize_object_action_node" name="initialize_object_action_server" />

  <node if="$(arg debug)" pkg="task_imitation" type="program_executor" name="program_executor" launch-prefix="xterm -e gdb --args" required="true" output="screen" />
  <node unless="$(arg debug)" pkg="task_imitation" type="program_executor" name="program_executor" required="true" output="screen" />
</launch>
